src/library_exports.js

```diff
@@ -0,0 +1,35 @@
+mergeInto(LibraryManager.library, {
+  emscripten_get_exported_function: function(name) {
+    name = UTF8ToString(name);
+#if WASM_BACKEND
+    // Wasm backend does not use C name mangling on exports,
+    // so adjust for that manually.
+    if (name[0] == '_') name = name.substr(1);
```

Comment:
I may be wrong, but I think this is reversed: in the wasm backend, the wasm exports look like `main`. We only add `_` when we place them on `Module` for exporting to JS, and you're not looking at that?

Reply:
I cannot look at `Module['_foo']` here because that is a JavaScript function that looks like
```js
var _foo = Module["_foo"] = function() {
  assert(runtimeInitialized, 'you need to wait for the runtime to be ready (e.g. wait for main() to be called)');
  assert(!runtimeExited, 'the runtime was exited (use NO_EXIT_RUNTIME to keep it alive after main() exits)');
  return Module["asm"]["foo"].apply(null, arguments)
};
```
and that would result in an extra trampoline, but also addFunction would yell "Assertion failed: Missing signature argument to addFunction". Hence looking at `Module["asm"]["foo"]` directly instead of `Module["_foo"]`.

Reply:
I see, thanks.